cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -209,92 +186,77 @@ def update_single_sample_read_count(self, sample_id: str, q30_threshold: int) ->
-                self.add_bundle_and_version_if_non_existent(bundle_name=sample_id)
+        valid_sample_fastq_paths = get_valid_sample_fastq_paths(flow_cell.path)
 
-                self.add_file_to_bundle_if_non_existent(
-                    file_path=fastq_file_path,
-                    bundle_name=sample_id,
-                    tag_names=[SequencingFileTag.FASTQ, flow_cell_name],
+        for sample_fastq_path in valid_sample_fastq_paths:
+            if self.fastq_path_should_be_stored_in_housekeeper(
+                sample_fastq_path=sample_fastq_path,
+                sequencer_type=flow_cell.sequencer_type,
+                flow_cell_name=flow_cell.id,
+            ):
+                self.store_fastq_path_in_housekeeper(
+                    sample_fastq_path=sample_fastq_path, flow_cell_name=flow_cell.id
                 )
 
-    def add_sample_sheet(self, flow_cell_directory: Path, flow_cell_name: str) -> None:
-        """Add sample sheet path to Housekeeper."""
-
-        sample_sheet_file_path: Path = self.find_sample_sheet_path(
-            flow_cell_directory=flow_cell_directory
-        )
+    def store_fastq_path_in_housekeeper(self, sample_fastq_path: Path, flow_cell_name: str) -> None:
+        sample_id = get_sample_id_from_sample_fastq(sample_fastq_path)
 
+        self.add_bundle_and_version_if_non_existent(bundle_name=sample_id)
         self.add_file_to_bundle_if_non_existent(
-            file_path=sample_sheet_file_path,
-            bundle_name=flow_cell_name,
-            tag_names=[SequencingFileTag.SAMPLE_SHEET, flow_cell_name],
+            file_path=sample_fastq_path,
+            bundle_name=sample_id,
+            tag_names=[SequencingFileTag.FASTQ, flow_cell_name],
         )
 
-    def find_sample_sheet_path(self, flow_cell_directory: Path):
+    def fastq_path_should_be_stored_in_housekeeper(
+        self, sample_fastq_path: Path, sequencer_type: Sequencers, flow_cell_name: str
+    ) -> bool:
         """
-        Recursively searches for the given sample sheet file in the provided flow cell directory.
-
-        Raises:
-            FileNotFoundError: If the sample sheet file is not found in the flow cell directory.
+        Check if a sample fastq file should be tracked in Housekeeper.
+        Only fastq files that pass the q30 threshold should be tracked.
         """
-        for directory_path, _, files in os.walk(flow_cell_directory):
-            if DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME in files:
-                LOG.info(f"Found sample sheet in {directory_path}")
-                return Path(directory_path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME)
-
-        raise FileNotFoundError(
-            f"Sample sheet not found in given flow cell directory: {flow_cell_directory}"
+        sample_id = get_sample_id_from_sample_fastq(sample_fastq_path)
+        lane = get_lane_from_sample_fastq(sample_fastq_path)
+        q30_threshold: int = get_q30_threshold(sequencer_type)
```

Comment:
I will leave this as is since I think `get_q30_threshold_for_sequencer_type` is too verbose and it should be clear from the argument name that it is from the sequencer type

Reply:
Not sure though, just a preference